First draft of CIDR documentation

ivpanda opened the pull request:
We don't currently have any documentation to help customers choose CIDR ranges, and they often get caught up on this aspect of networking. This document is meant to be a brief overview of what CIDR ranges are and how users can choose the appropriate ones.

I haven't yet added any images or screenshots; I can do that when the document is reviewed, so as to save us creating any unnecessary assets!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/redpanda-data/documentation?pullRequest=1546) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/redpanda-data/documentation?pullRequest=1546) before we can accept your contribution.<br/><hr/>**Ivana Bilic** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/redpanda-data/documentation?pullRequest=1546) it.</sub>

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,73 @@
+---
+title: Choosing CIDR ranges 
```
If this is a vital part/prereq for setting up VPC Peering, would it not be best placed as the first section in the VPC Peering topic? The current placement suggests it is a general RP Cloud topic. Given that VPC Peering falls under Dedicated Cloud, seems like that would be the context in which customers would look for/find this most useful, no? 
I notice the prereqs are identical for those setting up VPC Peering. If you were to place this content as the first heading in that topic, you could simply remove that section.

ivpanda replied:
I was struggling with where to put it. This is _part_ of the VPC peering process, which takes place in both Dedicated and Cloud. The VPC peering document that exists is for the VPC process within dedicated cloud only (there is no Redpanda guidance for VPC peering with BYOC; I'd like to write that soon enough as well). Where do you think this should live?

AnneLeeperRP replied:
My two cents:
I think this topic is explaining how to choose a CIDR range for the Redpanda network if you plan to use VPC peering. This is a step in the cluster creation workflow in the Cloud UI that needs to be documented more fully.

I think the best place for this material is in the "Create a Cluster" topics. We currently have one topic for creating a Dedicated cluster on AWS, and we have two topics for creating a BYOC cluster (one for AWS and one for GCP). See steps 10 and 11 in each topic, where we touch on the networking step (but we definitely need more detail and better guidance, which this PR addresses).

This PR brings up the requirements for choosing a CIDR range _if_ you choose **Private - VPC Peering** as the **Connection Type**:
* The Redpanda network (VPC) and the cloud provider VPC must be in the same region
* The IP addresses for _all_ VPCs in the region must not overlap with the IP addresses for the Redpanda VPC.

It explains how to find the CIDR ranges for the user's VPCs, and also links to a tool for calculating overlapping IP addresses, which will be helpful for users.

I don't think we need to explain CIDR notation here, since our users probably understand it already. If not, they can find lots of resources online to explain the concepts.

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,73 @@
+
+## Prerequisites
+
+* **VPC network** - Before you set up a peering connection in the Redpanda Cloud UI, you must have a VPC in your own account for Redpanda's VPC to connect to. If you do not already have a VPC, log in to the AWS VPC Console and create one.
+* **Matching Region** - VPC peering connections can only be established between your and Redpanda VPC networks created in the **same region**. We don't support Inter-Region VPC peering connections.
```
See suggestion above for moving this topic from the top level of the navtree into VPC Peering topic (then you could remove this Prereq section)

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,73 @@
+    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/cloud/cidr-ranges/" />
+</head>
+
+Choosing CIDR ranges is an essential part of the VPC peering process, required to ensure that data will be transferred between the networks. Choosing incorrect CIDRs means that the connection between your cloud provider and Redpanda will be unsuccessful. However, there are some pitfalls in this process, and this document will provide general guidance on how to choose the correct CIDR range for Re ...
```
```suggestion
Choosing Classless Inter-Domain Routing (CIDR) ranges is an essential part of the VPC peering process, required to ensure that data will be transferred between the networks. Choosing incorrect CIDRs means that the connection between your cloud provider and Redpanda will be unsuccessful. However, there are some pitfalls in this process, and this document will provide general guidance on how to choose the correct CIDR range for Redpanda.
```

ivpanda commented:
@Feediver1 Thank you! A few follow-ups after this review: 

1. Will this also have an engineering review? I think it might need one, just to make sure.
2. The screenshots: I had been meaning to add screenshots of AWS, not Redpanda. Does the guideline cover all screenshots as a general rule, or just Redpanda ones?

Feediver1 commented:
> @Feediver1 Thank you! A few follow-ups after this review:
> 
> 1. Will this also have an engineering review? I think it might need one, just to make sure.

All edits to docs that add or change technical information require an eng/SME review an approval. Short answer--yes. Please add a cloud engineer who works on VPC as a reviewer.
> 2. The screenshots: I had been meaning to add screenshots of AWS, not Redpanda. Does the guideline cover all screenshots as a general rule, or just Redpanda ones?
Covers all screenshots.

malinskibeniamin commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
````diff
@@ -0,0 +1,73 @@
+
+The output should look something like this: 
+
+```
````
For correct formatting (within Cloud UI) we should always use
```suggestion
```bash
```

malinskibeniamin commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,73 @@
+* **Matching Region** - VPC peering connections can only be established between your and Redpanda VPC networks created in the **same region**. We don't support Inter-Region VPC peering connections.
+
+:::tip
+Consider adding `rp` at the beginning of the VPC name to indicate that this VPC is for deploying a Redpanda cluster.
```
In Cloud UI we usually say `rp-` prefix

malinskibeniamin commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,73 @@
+
+## How to choose CIDR ranges
+
+To choose a range for Redpanda, you'll need to know your AWS VPC CIDR first. You can find its CIDR range in [the VPC console in AWS](https://console.aws.amazon.com/vpc/). There, you’ll be able to see your list of VPCs:
```
Should we differentiate between AWS and GCP

ivpanda replied:
Ah yeah, I used AWS in this example but that's not very clear; I'm assuming that the technical details between the two don't really matter for the purposes of this help article? I can change this to remove "AWS" references and just say "cloud provider CIDR" more generically?

ivpanda commented:
@Feediver1 — I've updated this to be less about AWS and more about general cloud providers. I've also added another example.

I'm still not sure where to put this because I feel as though it's general (if supplemental) "Set up peering" documentation, whether it's a BYOC or dedicated cluster. Putting it as a subtopic to VPC peering would make it harder to find, I'm assuming, because it would be under BYOC and then again under VPC Peering. 

 @malinskibeniamin — I've added a couple of examples here, a check for correctness would be appreciated!

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+    <link rel="canonical" href="https://docs.redpanda.com/docs/deploy/deployment-option/cloud/cidr-ranges/" />
+</head>
+
+Choosing Classless Inter-Domain Routing (CIDR) ranges is an essential part of the VPC peering process, required to ensure that data will be transferred between the networks. Choosing incorrect CIDRs means that the connection between your cloud provider and Redpanda will be unsuccessful. However, there are some pitfalls in this process, and this document will provide general guidance on how to cho ...
```
```suggestion
Choosing Classless Inter-Domain Routing (CIDR) ranges is an essential part of the VPC peering process, required to ensure that data will be transferred between the networks. Choosing incorrect CIDRs means that the connection between your cloud provider and Redpanda will be unsuccessful. This topic provides guidance for selecting the correct CIDR range for Redpanda.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+Choosing Classless Inter-Domain Routing (CIDR) ranges is an essential part of the VPC peering process, required to ensure that data will be transferred between the networks. Choosing incorrect CIDRs means that the connection between your cloud provider and Redpanda will be unsuccessful. However, there are some pitfalls in this process, and this document will provide general guidance on how to cho ...
+
+:::note
+This document is a set of general recommendations for choosing a CIDR range to peer. If you have a complex networking setup, it's best to ask a networking engineer which Redpanda CIDR would not conflict with your existing VPCs.
```
```suggestion
This topic provides a set of general recommendations for choosing a CIDR range to peer. If you have a complex networking setup, it's recommended that you work with a networking engineer to identify which Redpanda CIDRs would not conflict with your existing VPCs.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+
+## Prerequisites
+
+* **VPC network** - Before you set up a peering connection in the Redpanda Cloud UI, you must have another VPC for Redpanda to connect to. If you do not already have a network, create one in your cloud provider.
+* **Matching Region** - VPC peering connections can only be established between your and Redpanda VPC networks created in the **same region**. We don't support Inter-Region VPC peering connections.
```
```suggestion
* **VPC network** - Before setting up a peering connection in Redpanda Cloud, you must have another VPC to which Redpanda can connect. If you do not already have a network, create one in your cloud provider.
* **Matching Region** - VPC peering connections can only be established between your and Redpanda VPC networks created in the **same region**. Redpanda Cloud does not support Inter-Region VPC peering connections.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+1. Your existing (client) VPC CIDR 
+2. Your Redpanda cluster CIDR  
+
+However, finding these ranges and ensuring they don’t overlap when setting up VPC peering can sometimes be difficult. 
```
```suggestion
Finding these ranges and ensuring that they don’t overlap when setting up VPC peering can sometimes be a challenge. 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+
+However, finding these ranges and ensuring they don’t overlap when setting up VPC peering can sometimes be difficult. 
+
+## How to choose CIDR ranges
```
```suggestion
## Choose the CIDR ranges
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+
+## How to choose CIDR ranges
+
+To choose a range for Redpanda, you'll need to know your VPC CIDR first. You can find its this in [the VPC console in AWS](https://console.aws.amazon.com/vpc/), where it will be labelled **IPv4 CIDRs** or in the Details view of your VPC in GCP, where they're labelled **Internal IP Ranges**.
```
The AWS link here take users to the following screen:
![image](https://user-images.githubusercontent.com/102751339/236315091-be8b5028-fe03-4f63-865e-3472df57137a.png)

Feediver1 replied:
Perhaps you can link them to the AWS doc on the VPC console, which would guide them? Otherwise, this is not a good link/user experience. Also, you mention GCP, but no link is provided. So there's that inconsistency.

ivpanda replied:
I've removed the link and tried to be specific about where to find it. The AWS documentation doesn't really provide an appropriate link, either.

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+
+## What are CIDRs? 
+
+CIDR ranges are a critical part of Redpanda’s BYOC setup process. There are two CIDR values that are important: 
```
```suggestion
CIDR ranges are a critical part of Redpanda’s BYOC setup process. Two important CIDR values are: 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+
+To choose a range for Redpanda, you'll need to know your VPC CIDR first. You can find its this in [the VPC console in AWS](https://console.aws.amazon.com/vpc/), where it will be labelled **IPv4 CIDRs** or in the Details view of your VPC in GCP, where they're labelled **Internal IP Ranges**.
+
+You can check which IPs this range encompasses by using the [ipcalc](https://www.linux.com/topic/networking/how-calculate-network-addresses-ipcalc/) command in your Terminal or the [CIDR Calculation tool](https://www.ipaddressguide.com/cidr). For example, if your client's CIDR range is 10.0.0.0/20, then you can run the following command: 
```
```suggestion
You can check which IPs this range encompasses by using the [ipcalc](https://www.linux.com/topic/networking/how-calculate-network-addresses-ipcalc/) command in your terminal or the [CIDR Calculation tool](https://www.ipaddressguide.com/cidr). For example, if your client's CIDR range is 10.0.0.0/20, then run the following command: 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
````diff
@@ -0,0 +1,75 @@
+Hosts/Net: 4094                  Class A, Private Internet
+``` 
+
+Note `HostMin` (10.0.0.1) and `HostMax` (10.0.15.254). These two values are the minimum and maximum of the range of 4,094 IPs that this CIDR covers. The number of IPs is governed by the suffix: /16 contains 65534 IPs, /21 contains 2046, /24 contains 254, and so on. For private networks, this number can range from 8 (which contains 16777214 IPs) and 30 (which contains 2). 
````
```suggestion
Note the values for `HostMin` (10.0.0.1) and `HostMax` (10.0.15.254); these are the minimum and maximum of the range of 4,094 IPs that this CIDR covers. The number of IPs is governed by the suffix: /16 contains 65534 IPs, /21 contains 2046, /24 contains 254, and so on. For private networks, this number can range from 8 (which contains 16777214 IPs) and 30 (which contains 2). 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+Note `HostMin` (10.0.0.1) and `HostMax` (10.0.15.254). These two values are the minimum and maximum of the range of 4,094 IPs that this CIDR covers. The number of IPs is governed by the suffix: /16 contains 65534 IPs, /21 contains 2046, /24 contains 254, and so on. For private networks, this number can range from 8 (which contains 16777214 IPs) and 30 (which contains 2). 
+
+:::note
+When defining your Redpanda CIDR, we require a block size between /16 and /20.
```
```suggestion
Your Redpanda CIDR requires a block size between /16 and /20.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+
+## Example 
+
+If we assume that your client's CIDR range is `10.0.0.0/20`, then your Redpanda range cannot overlap with it; if it does, VPC peering will not work. 
```
```suggestion
Assuming that your client's CIDR range is `10.0.0.0/20`, then your Redpanda range cannot overlap with it; if it does, VPC peering will not work. 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+## Example 
+
+If we assume that your client's CIDR range is `10.0.0.0/20`, then your Redpanda range cannot overlap with it; if it does, VPC peering will not work. 
+A limited set of examples that would work with `10.0.0.0/20` would be `10.8.0.0/20`, `10.0.16.0/20`, or `10.1.0.0/20`. What would not work would be something like `10.0.0.6/20`, `10.0.8.0/20`, and `10.0.1.7/20`. [This](http://trk.free.fr/ipcalc/tools.html) is a good tool for checking overlapping IPs. 
```
```suggestion
A limited set of examples that would work with `10.0.0.0/20` would be `10.8.0.0/20`, `10.0.16.0/20`, or `10.1.0.0/20`. Ranges like `10.0.0.6/20`, `10.0.8.0/20`, or `10.0.1.7/20` would not work. [This](http://trk.free.fr/ipcalc/tools.html) is a good tool for checking overlapping IPs. 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+If we assume that your client's CIDR range is `10.0.0.0/20`, then your Redpanda range cannot overlap with it; if it does, VPC peering will not work. 
+A limited set of examples that would work with `10.0.0.0/20` would be `10.8.0.0/20`, `10.0.16.0/20`, or `10.1.0.0/20`. What would not work would be something like `10.0.0.6/20`, `10.0.8.0/20`, and `10.0.1.7/20`. [This](http://trk.free.fr/ipcalc/tools.html) is a good tool for checking overlapping IPs. 
+
+`10.0.0.6/20`, `10.0.0.7/20`, and `10.0.1.7/20`
```
```suggestion

```
Removed values here ^^ because they were hanging on their own, contextless.

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+`10.0.0.6/20`, `10.0.0.7/20`, and `10.0.1.7/20`
+## Multi-VPC example 
+
+If you have many IP ranges allocated in a more complex system, it’s probably best to speak with a network engineer who can help with IP allocation. Your Redpanda CIDR cannot overlap with any of your existing VPCs, not just the one you want to peer with.
```
```suggestion
If you have many IP ranges allocated in a more complex system, it's recommended that you seek guidance from a network engineer who can help with IP allocation. Your Redpanda CIDR cannot overlap with any of your existing VPCs, not just the one you want to peer with.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,75 @@
+
+If you have many IP ranges allocated in a more complex system, it’s probably best to speak with a network engineer who can help with IP allocation. Your Redpanda CIDR cannot overlap with any of your existing VPCs, not just the one you want to peer with.
+
+If the following example ranges are in use: `10.0.0.0/20`, `10.8.0.0/20`, `10.0.35.8/20`, and `10.0.16.8/20`. 
+
+Then a Redpanda CIDR that would work (and not overlap) with any of them would be `10.8.48.8/20`. 
```
```suggestion
If the following example ranges are in use: 

- `10.0.0.0/20`
- `10.8.0.0/20`
- `10.0.35.8/20`
- `10.0.16.8/20` 

then a Redpanda CIDR that would work (and not overlap) with any of them would be `10.8.48.8/20`. 
```

ivpanda commented:
@Feediver1 just checking in— is there anything I need to do to move this forward and merge? e

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+Choosing Classless Inter-Domain Routing (CIDR) ranges is an essential part of the VPC peering process, required to ensure that data will be transferred between the networks. Choosing incorrect CIDRs means that the connection between your cloud provider and Redpanda will be unsuccessful. This topic provides guidance for selecting the correct CIDR range for Redpanda.
+
+:::note
+This topic provides a set of general recommendations for choosing a CIDR range to peer. If you have a complex networking setup, it's recommended that you work with a networking engineer to identify which Redpanda CIDRs would not conflict with your existing VPCs.
```
```suggestion
This topic provides a set of general recommendations for choosing a CIDR range to peer. If you have a complex networking setup, it's recommended that you work with a networking engineer to identify the Redpanda CIDRs that do not conflict with your existing VPCs.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+## Prerequisites
+
+* **VPC network** - Before setting up a peering connection in Redpanda Cloud, you must have another VPC to which Redpanda can connect. If you do not already have a network, create one in your cloud provider.
+* **Matching Region** - VPC peering connections can only be established between your and Redpanda VPC networks created in the **same region**. Redpanda Cloud does not support Inter-Region VPC peering connections.
```
```suggestion
* **Matching Region** - VPC peering connections can only be established between your and Redpanda VPC networks created in the _same region_. Redpanda Cloud does not support Inter-Region VPC peering connections.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+* **Matching Region** - VPC peering connections can only be established between your and Redpanda VPC networks created in the **same region**. Redpanda Cloud does not support Inter-Region VPC peering connections.
+
+:::tip
+Consider adding an `rp-` prefix at the beginning of the VPC name to indicate that this VPC is for deploying a Redpanda cluster.
```
```suggestion
Consider adding an `rp-` prefix to the VPC name to indicate that the VPC is for deploying a Redpanda cluster.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+1. Your existing (client) VPC CIDR 
+2. Your Redpanda cluster CIDR  
+
+Finding these ranges and ensuring that they don’t overlap when setting up VPC peering can sometimes be a challenge. 
```
```suggestion
While it may be a challenge to identify these ranges, you must ensure that that they don’t overlap when setting up VPC peering. 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+
+## Choose the CIDR ranges
+
+To choose a range for Redpanda, you'll need to know your VPC CIDR first. You can find its this in the VPC area of the AWS Management Console, where it will be labelled **IPv4 CIDRs**. In GCP, it will be in the Details view of your VPC, where they're labelled **Internal IP Ranges**.
```
```suggestion
To choose a range for Redpanda, you must know your VPC CIDR, which you can find in the VPC area of the AWS Management Console, labeled **IPv4 CIDRs**. In GCP, it is located in the Details view of your VPC, and labeled **Internal IP Ranges**.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
````diff
@@ -0,0 +1,79 @@
+Hosts/Net: 4094                  Class A, Private Internet
+``` 
+
+Note the values for `HostMin` (10.0.0.1) and `HostMax` (10.0.15.254); these are the minimum and maximum of the range of 4,094 IPs that this CIDR covers. The number of IPs is governed by the suffix: /16 contains 65534 IPs, /21 contains 2046, /24 contains 254, and so on. For private networks, this number can range from 8 (which contains 16777214 IPs) and 30 (which contains 2). 
````
```suggestion
Note the values for `HostMin` (10.0.0.1) and `HostMax` (10.0.15.254); these are the minimum and maximum of the range of 4,094 IPs that this CIDR covers. The number of IPs is governed by the suffix: /16 contains 65534 IPs, /21 contains 2046, /24 contains 254, and so on. For private networks, this number can range from 8 (which contains 16777214 IPs) to 30 (which contains 2). 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+## Example 
+
+Assuming that your client's CIDR range is `10.0.0.0/20`, then your Redpanda range cannot overlap with it; if it does, VPC peering will not work. 
+A limited set of examples that would work with `10.0.0.0/20` would be `10.8.0.0/20`, `10.0.16.0/20`, or `10.1.0.0/20`. Ranges like `10.0.0.6/20`, `10.0.8.0/20`, or `10.0.1.7/20` would not work. [ipcalc](http://trk.free.fr/ipcalc/tools.html) is a good tool for checking overlapping IPs. 
```
```suggestion
A limited set of examples that would work with `10.0.0.0/20` would be `10.8.0.0/20`, `10.0.16.0/20`, or `10.1.0.0/20`. Ranges like `10.0.0.6/20`, `10.0.8.0/20`, or `10.0.1.7/20` would not work. You can use [ipcalc](http://trk.free.fr/ipcalc/tools.html) to check for overlapping IPs. 
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+
+## Multi-VPC example 
+
+If you have many IP ranges allocated in a more complex system, it's recommended that you seek guidance from a network engineer who can help with IP allocation. Your Redpanda CIDR cannot overlap with any of your existing VPCs, not just the one you want to peer with.
```
```suggestion
If you have many IP ranges allocated in a complex system, it's recommended that you seek guidance from a network engineer who can help with IP allocation. Your Redpanda CIDR cannot overlap with any of your existing VPCs, nor can it overlap with the VPC you want to peer with.
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+
+If you have many IP ranges allocated in a more complex system, it's recommended that you seek guidance from a network engineer who can help with IP allocation. Your Redpanda CIDR cannot overlap with any of your existing VPCs, not just the one you want to peer with.
+
+If the following example ranges are in use: 
```
```suggestion
If the following example ranges are in use, then a Redpanda CIDR that would work (and not overlap) with any of them would be `10.8.48.8/20`:
```

Feediver1 commented on docs/deploy/deployment-option/cloud/cidr-ranges.mdx:
```diff
@@ -0,0 +1,79 @@
+- `10.0.35.8/20`
+- `10.0.16.8/20` 
+
+then a Redpanda CIDR that would work (and not overlap) with any of them would be `10.8.48.8/20`. 
```
```suggestion

```

Feediver1 commented on docs/deploy/deployment-option/cloud/networking/index.mdx:
```diff
@@ -0,0 +1,20 @@
+
+  - [Choosing CIDR ranges](../../cloud/cidr-ranges)
+  
+    Guidelines for choosing CIDR ranges when creating a cluster 
```
```suggestion
    Guidelines for choosing CIDR ranges when creating a cluster. 
```
